andela/hc-bigas-opus
# Assigning check priorities #155965584

# What does this PR do?

* Add priorities to check's in a users dashboard, the highest priority is displayed the first one.

## Description of Task to be completed?

* Choose a priority for a check in 5 levels, `Highest` being the top most, `Lowest` being the last. the check with the highest priority is the first one in the list

## How should this be manually tested?

* Open health checks.io on your localhost
* Navigate to the checks page.

* By default all priorities are normal, click on the normal button, a modal will appear with a slider and select form.
* Choose the level appropriate for the check priority, then save it to update the check's priority level.

### The files changed/created include:

##### Edited

* hc/api/models.py - Added priority column and function
* hc/front/forms.py - Added Class priority form
* hc/front/urls.py - Added the URL that redirects to priority modal form
* hc/front/views.py - Import and added priority as the main filter for checks
* hc/api/tests/test_notify.py - Comment out failing SMS test
* static/js/checks.js - Add functions form manipulating data in the slider depending on the priority level
* templates/front/my_checks.html -  Added the modal responsible for counting the priority form
* templates/front/my_checks_desktop.html - Added Normal priority as the default value for all checks created

#### What are the relevant pivotal tracker stories?
* [#155965584](https://www.pivotaltracker.com/story/show/155965584)

#### screenshots

<img width="1280" alt="1" src="https://user-images.githubusercontent.com/15923109/38363746-8c6c8db4-38de-11e8-9d15-6c7a01a9d528.png">
<img width="1280" alt="2" src="https://user-images.githubusercontent.com/15923109/38363747-8cad71e4-38de-11e8-8026-c08a957ba450.png">
<img width="1280" alt="3" src="https://user-images.githubusercontent.com/15923109/38363748-8cd9cd84-38de-11e8-8243-d67c4257ec74.png">
<img width="1280" alt="4" src="https://user-images.githubusercontent.com/15923109/38363749-8d079a98-38de-11e8-8096-1848aa62a70d.png">
<img width="1280" alt="5" src="https://user-images.githubusercontent.com/15923109/38363751-8d331826-38de-11e8-92c3-cf4801745d8c.png">
